components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/ext/APIManagerExtensionHandler.java

```diff
@@ -71,9 +74,25 @@ public boolean mediate(MessageContext messageContext, String direction) {
     public boolean handleRequest(MessageContext messageContext) {
         Timer.Context context = startMetricTimer(DIRECTION_IN);
         long executionStartTime = System.nanoTime();
+        TracingSpan requestMediationSpan = null;
+        if (Util.tracingEnabled()) {
+            TracingSpan responseLatencySpan =
+                    (TracingSpan) messageContext.getProperty(APIMgtGatewayConstants.RESPONSE_LATENCY);
+            TracingTracer tracer = Util.getGlobalTracer();
+            requestMediationSpan = Util.startSpan(APIMgtGatewayConstants.REQUEST_MEDIATION, responseLatencySpan, tracer);
+        }
         try {
             return mediate(messageContext, DIRECTION_IN);
+        } catch (Exception e) {
```

Comment:
Can't we catch specific exception, instead of catching generic exception here

Reply:
I'm trying catch exception whether it is there or not in general. So that I am able to set the tag in open tracing UI. eg: Zipkin